### PR TITLE
Follow-up Mail app

### PR DESCRIPTION
**Pull request by EDsCODE**
- created a mail app that holds migration, model, and management command that will be called by heroku scheduler to on an interval to check for mails that should be sent
- we can use this to add a bunch of triggers/commands that should send emails to people who sign up/need a follow up

Todo::
- [x] figure out more optimal way (don't need to query all teams everytime this job runs)
- [x] relies on this [pr](https://github.com/PostHog/posthog/pull/1075)
~- [ ] Set up domain verification for Sendgrid / change from email (currently it's just eric@posthog)~
~- [ ] add Sendgrid API key to heroku deployment~ 
- [ ] add scheduler job after deployed

### CODE REVIEW

File: mail_app/management/commands/follow_up_mailer.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            self._send_follow_up_to_user(user.email)
+        
+        FollowUpEmail.objects.create(team_id=team.pk)
+    
+    def _send_follow_up_to_user(self, email: str):
+        message = Mail(
+            from_email='eric@posthog.com',
+            to_emails=email,
+            subject='Follow Up Email',
+            html_content='Looks like you haven\'t started sending events with Posthog yet. Want a demo?')
```
Comment (timgl) [inline mail_app/management/commands/follow_up_mailer.py:38]: Do you want to merge like this or do you want me to write an email for this?

**Comment by timgl** [comment]
I think this looks good! [This article](https://medium.com/@DoorDash/tips-for-building-high-quality-django-apps-at-scale-a5a25917b2b5) does warn against using separate Django applications for different bits of functionality (and echoes my experience at Arachnys) but this is also a super small experiment so happy either way

**Comment by EDsCODE** [comment]
Thanks for the article! I guess I should be more on the side "I'm not sure about apps so avoid them." However, it seemed like the most seamless way to do the copy pasting into main repo. I deliberately removed the foreign key dependency so querying for teams for FollowUpEmail is just looking at a number. Also, apps seem to help encapsulate into microservice architecture as mentioned in the article and the mailer functions probably should be a microservice at scale!

File: mail_app/management/commands/follow_up_mailer.py (hunk 1 above)
Comment (EDsCODE) [reply mail_app/management/commands/follow_up_mailer.py]: I made another edit so you can give it a look! I think by default it's sending from your email now so you should change it to however you would like it phrased. I tried to make it super casual so it sounds somewhat like we're actually personally emailing

**Comment by fuziontech** [comment]
Other than the above this looks awesome! We should get this in so that we can start sending these nudges to users.

Something to keep in mind is some people will get mad that there is no 'unsubscribe' button on the email so we should fast follow with a feature for unsubscribing...or we just push this to a 3rd party service like mailchimp

**Comment by Twixes** [comment]
@EDsCODE @timgl @fuziontech Any more work to be done here?

**Comment by paolodamico** [comment]
Closing this in favor of #20

**Pull request closed without merging**